cmd/vm-informant/main.go

```diff
@@ -103,48 +110,94 @@ func main() {
 // We execute ourselves as a subprocess so that it's possible to appropriately cleanup after
 // termination by various signals (or an unhandled panic!). This is worthwhile because we *really*
 // don't want to leave the cgroup frozen while waiting to restart.
-func runRestartOnFailure(args []string, cleanupHooks []func()) {
+func runRestartOnFailure(ctx context.Context, args []string, cleanupHooks []func()) {
 	selfPath := os.Args[0]
-
-	minWaitDuration := time.Second * 5
+	timer := time.NewTimer(0)
+	defer timer.Stop()
+
+	// CancelFunc for each process's context, for internal signaling.
+	// The defer is out here so we don't leak deferred funcs on restart.
```

Comment:
I don't think this matters, particularly: 
- the previous defer was in a function that started on line 129, which meant that it always had run by at least line 174, which is before before the next iteration of the loop, under any branch. Though I think the previous code was a bit safer to edit, on the case that someone adds an early return to the 129 function that isn't the cause of the outer context being canceled? 
- the only place where this could matter is if the goroutine on 139 panics (which isn't handled right now, probably correctly) or if if something in the function on line 129 panics, (which, except for the cleanup hook(s) is just `os/exec`) so I think we're firmly in "moving the deck chairs on the titanic" territory.

Reply:
Shoot yeah, right. Missed that it was in an inner function already.